diff -Naur a/png.c b/png.c
--- a/png.c
+++ b/png.c
@@ -13,6 +13,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 /* Generate a compiler error if there is an old png.h in the search path. */
 typedef png_libpng_version_1_6_37 Your_png_h_is_not_version_1_6_37;
 
diff -Naur a/pngerror.c b/pngerror.c
--- a/pngerror.c
+++ b/pngerror.c
@@ -18,6 +18,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #if defined(PNG_READ_SUPPORTED) || defined(PNG_WRITE_SUPPORTED)
 
 static PNG_FUNCTION(void, png_default_error,PNGARG((png_const_structrp png_ptr,
diff -Naur a/pngget.c b/pngget.c
--- a/pngget.c
+++ b/pngget.c
@@ -14,6 +14,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #if defined(PNG_READ_SUPPORTED) || defined(PNG_WRITE_SUPPORTED)
 
 png_uint_32 PNGAPI
diff -Naur a/pngmem.c b/pngmem.c
--- a/pngmem.c
+++ b/pngmem.c
@@ -19,6 +19,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #if defined(PNG_READ_SUPPORTED) || defined(PNG_WRITE_SUPPORTED)
 /* Free a png_struct */
 void /* PRIVATE */
diff -Naur a/pngpread.c b/pngpread.c
--- a/pngpread.c
+++ b/pngpread.c
@@ -13,6 +13,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_PROGRESSIVE_READ_SUPPORTED
 
 /* Push model modes */
diff -Naur a/pngread.c b/pngread.c
--- a/pngread.c
+++ b/pngread.c
@@ -19,6 +19,10 @@
 #  include <errno.h>
 #endif
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_READ_SUPPORTED
 
 /* Create a PNG structure for reading, and allocate any memory needed. */
diff -Naur a/pngrio.c b/pngrio.c
--- a/pngrio.c
+++ b/pngrio.c
@@ -20,6 +20,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_READ_SUPPORTED
 
 /* Read the data from whatever input you are using.  The default routine
diff -Naur a/pngrtran.c b/pngrtran.c
--- a/pngrtran.c
+++ b/pngrtran.c
@@ -18,6 +18,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_ARM_NEON_IMPLEMENTATION
 #  if PNG_ARM_NEON_IMPLEMENTATION == 1
 #    define PNG_ARM_NEON_INTRINSICS_AVAILABLE
diff -Naur a/pngrutil.c b/pngrutil.c
--- a/pngrutil.c
+++ b/pngrutil.c
@@ -16,6 +16,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_READ_SUPPORTED
 
 png_uint_32 PNGAPI
diff -Naur a/pngset.c b/pngset.c
--- a/pngset.c
+++ b/pngset.c
@@ -18,6 +18,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #if defined(PNG_READ_SUPPORTED) || defined(PNG_WRITE_SUPPORTED)
 
 #ifdef PNG_bKGD_SUPPORTED
diff -Naur a/pngtrans.c b/pngtrans.c
--- a/pngtrans.c
+++ b/pngtrans.c
@@ -13,6 +13,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #if defined(PNG_READ_SUPPORTED) || defined(PNG_WRITE_SUPPORTED)
 
 #if defined(PNG_READ_BGR_SUPPORTED) || defined(PNG_WRITE_BGR_SUPPORTED)
diff -Naur a/pngwio.c b/pngwio.c
--- a/pngwio.c
+++ b/pngwio.c
@@ -20,6 +20,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_WRITE_SUPPORTED
 
 /* Write the data to whatever output you are using.  The default routine
diff -Naur a/pngwrite.c b/pngwrite.c
--- a/pngwrite.c
+++ b/pngwrite.c
@@ -16,6 +16,10 @@
 #  include <errno.h>
 #endif /* SIMPLIFIED_WRITE_STDIO */
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_WRITE_SUPPORTED
 
 #ifdef PNG_WRITE_UNKNOWN_CHUNKS_SUPPORTED
diff -Naur a/pngwtran.c b/pngwtran.c
--- a/pngwtran.c
+++ b/pngwtran.c
@@ -13,6 +13,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_WRITE_SUPPORTED
 #ifdef PNG_WRITE_TRANSFORMS_SUPPORTED
 
diff -Naur a/pngwutil.c b/pngwutil.c
--- a/pngwutil.c
+++ b/pngwutil.c
@@ -13,6 +13,10 @@
 
 #include "pngpriv.h"
 
+#ifdef _MSC_VER
+#pragma warning(disable: 5045)
+#endif    /* _MSC_VER */
+
 #ifdef PNG_WRITE_SUPPORTED
 
 #ifdef PNG_WRITE_INT_FUNCTIONS_SUPPORTED
